src/api/java/com/minecolonies/api/util/MessageUtils.java

```diff
@@ -0,0 +1,153 @@
+    }
+
+    /**
+     * Send a message to the given player.
+     *
+     * @param player the player.
+     * @param colony the colony which sent the message.
+     * @param key    the translation component key.
+     * @param args   the additional args to send in the translation component.
+     */
+    public static void sendPlayerColonyMessage(PlayerEntity player, IColony colony, String key, Object... args)
+    {
+        sendPlayerColonyMessageWithColor(player, colony, TextFormatting.WHITE, new TranslationTextComponent(key, args));
+    }
+
+    /**
+     * Send a message to the given player.
+     *
+     * @param player    the player.
+     * @param colony    the colony which sent the message.
+     * @param component the message component.
+     */
+    public static void sendPlayerColonyMessage(PlayerEntity player, IColony colony, ITextComponent component)
+    {
+        sendPlayerColonyMessageWithColor(player, colony, TextFormatting.WHITE, component);
+    }
+
+    /**
+     * Send a message to the given player.
+     *
+     * @param player the player.
+     * @param colony the colony which sent the message.
+     * @param key    the translation component key.
+     * @param args   the additional args to send in the translation component.
+     */
+    public static void sendPlayerColonyMessageWithColor(PlayerEntity player, IColony colony, TextFormatting color, String key, Object... args)
+    {
+        sendPlayerColonyMessageWithColor(player, colony, color, new TranslationTextComponent(key, args));
+    }
+
+    /**
+     * Send a message to the given player.
+     *
+     * @param player    the player.
+     * @param colony    the colony which sent the message.
+     * @param component the message component.
+     */
+    public static void sendPlayerColonyMessageWithColor(PlayerEntity player, IColony colony, TextFormatting color, ITextComponent component)
+    {
+        ITextComponent fullComponent = component;
+
+        if (colony != null && !colony.isCoordInColony(player.level, player.blockPosition()))
+        {
+            fullComponent = new StringTextComponent("[" + colony.getName() + "] ")
+                              .withStyle(TextFormatting.WHITE)
+                              .append(component);
+        }
+
+        sendMessageInternal(player, fullComponent, Collections.singletonList(color));
+    }
+
+    /**
+     * Internal message sending handler.
+     *
+     * @param player     the player to send to.
+     * @param component  the text component to send.
+     * @param formatters a collection of formatter to apply to the root component.
+     */
+    private static void sendMessageInternal(PlayerEntity player, ITextComponent component, Collection<TextFormatting> formatters)
+    {
+        if (player == null)
+        {
+            Log.getLogger().log(Level.ERROR, "Attempting to send player message without passing a player, please report this to the developers!");
```

Comment:
This should crash in dev space (same way as in dist executor)

Reply:
I have not had this error just yet, a player technically does exist, he just can't be authenticated against mojang it's authorization servers, but the instance does exist.

Reply:
w/e applies to all if null checks in this class